haystack/environment.py

```diff
@@ -18,6 +18,12 @@
 env_meta_data: Dict[str, Any] = {}
 
 
+def set_pytorch_secure_model_loading(flag_val="1"):
+    # To load secure only model pytorch requires value of
+    # TORCH_FORCE_WEIGHTS_ONLY_LOAD to be ["1", "y", "yes", "true"]
```

Comment:
I think we should check whether this flag is set to any value different from `["1", "y", "yes", "true"]` before setting it to 1. Maybe a user explicitly set it to False. In that case we shouldn't silently overwrite it. Instead, let's **at least** log a warning.

Reply:
Changed it. I think your suggestion is taken care of, but please do let me know if you disagree.